### PR TITLE
added loadbalancer

### PR DESCRIPTION
**Pull request by geoffrey1330**

### CODE REVIEW

File: level2/loadbalancer.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+resource "aws_security_group" "lb-sg" {
+  name   = "${var.env_code}-lb-sg"
+  vpc_id = data.terraform_remote_state.networking.outputs.vpc_id #aws_vpc.main.id
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["105.112.100.127/32", "10.0.0.0/16"]
+  }
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf]: Port 22 is not needed

File: level2/loadbalancer.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["105.112.100.127/32", "10.0.0.0/16"]
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf]: "10.0.0.0/16" address is not needed

File: level2/loadbalancer.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+
+resource "aws_lb" "app-lb" {
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf]:
```suggestion
}

resource "aws_lb" "app-lb" {
```

File: level2/loadbalancer.tf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  security_groups    = [aws_security_group.lb-sg.id]
+  subnets            = [data.terraform_remote_state.networking.outputs.public-subnet_id,data.terraform_remote_state.networking.outputs.private-subnet_id]
+
+  enable_deletion_protection = true
+
+#   access_logs {
+#     bucket  = "israel-terraform"
+#     prefix  = "app-lb"
+#     enabled = true
+#   }
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf]: Don't leave commented-out code blocks

File: level2/loadbalancer.tf [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  tags = {
+    Environment = "production"
+  }
+}
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf]: please add trailing newline

File: level2/loadbalancer.tf [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+
+resource "aws_lb" "app-lb" {
+  name               = "${var.env_code}-lb"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.lb-sg.id]
+  subnets            = [data.terraform_remote_state.networking.outputs.public-subnet_id,data.terraform_remote_state.networking.outputs.private-subnet_id]
+
+  enable_deletion_protection = true
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf]: This options is not needed

File: level2/loadbalancer.tf [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    protocol    = "tcp"
+    cidr_blocks = ["105.112.100.127/32", "10.0.0.0/16"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf:17]: Please add proper tag

File: level2/loadbalancer.tf [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+#   access_logs {
+#     bucket  = "israel-terraform"
+#     prefix  = "app-lb"
+#     enabled = true
+#   }
+
+  tags = {
+    Environment = "production"
+  }
```
Comment (vladimirmukhin) [inline level2/loadbalancer.tf]: Please use proper tags that correspond to the naming strategy

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
This load balancer is not assigned to any of your instances.
You should assign the load balancer to your instance and make sure a web page is returned when you open load balancer url

File: level2/loadbalancer.tf (hunk 1 above)
Comment (geoffrey1330) [reply level2/loadbalancer.tf]: OK

File: level2/loadbalancer.tf (hunk 2 above)
Comment (geoffrey1330) [reply level2/loadbalancer.tf]: OK

File: level2/loadbalancer.tf (hunk 4 above)
Comment (geoffrey1330) [reply level2/loadbalancer.tf]: OK

File: level2/loadbalancer.tf (hunk 6 above)
Comment (geoffrey1330) [reply level2/loadbalancer.tf]: So i should remove it?

File: level2/loadbalancer.tf (hunk 7 above)
Comment (geoffrey1330) [reply level2/loadbalancer.tf:17]: Sorry to ask what do you mean by proper tag

File: level2/loadbalancer.tf (hunk 8 above)
Comment (geoffrey1330) [reply level2/loadbalancer.tf]: OK

File: level2/loadbalancer.tf (hunk 7 above)
Comment (vladimirmukhin) [reply level2/loadbalancer.tf:17]: One that has env_code in it

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**